meg_runtime/ui/addpluginpanel.py

```diff
@@ -0,0 +1,130 @@
+from PyQt5 import QtWidgets
+
+from meg_runtime.ui.manager import UIManager
+from meg_runtime.plugins import PluginManager
+from meg_runtime.ui.basepanel import BasePanel
+
+class AddPluginPanel(BasePanel):
+    """Setup the plugin panel."""
+
+    selected_file = None
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_title(self):
+        """Get the title of this panel."""
+        return 'Add New Plugin'
+
+    def on_load(self):
+        """Load dynamic elements within the panel."""
+
+        # Attach handlers
+        instance = self.get_widgets()
+        self.add_button = instance.findChild(QtWidgets.QPushButton, 'addButton')
+        self.add_button.clicked.connect(self.add_plugin)
+        # available plugin handlers
+        self.available_radio_button = instance.findChild(QtWidgets.QRadioButton, 'availableRadioButton')
+        self.available_radio_button.clicked.connect(self.enable_available_selection)
+        self.available_plugin_list = instance.findChild(QtWidgets.QTreeWidget, 'availablePluginList')
+        # file plugin handlers
+        self.file_radio_button = instance.findChild(QtWidgets.QRadioButton, 'fileRadioButton')
+        self.file_radio_button.clicked.connect(self.enable_file_selection)
+        self.choose_file_button = instance.findChild(QtWidgets.QPushButton, 'chooseFileButton')
+        self.choose_file_button.clicked.connect(self.open_file_dialog)
+        self.file_label = instance.findChild(QtWidgets.QLabel, 'fileLabel')
+        # url plugin handlers
+        self.url_radio_button = instance.findChild(QtWidgets.QRadioButton, 'urlRadioButton')
+        self.url_radio_button.clicked.connect(self.enable_url_selection)
+        self.url_field = instance.findChild(QtWidgets.QLineEdit, 'urlField')
+
+        # reset form values
+        self.available_radio_button.setChecked(True)
+        self.available_plugin_list.clear()
+        self.file_label.setText('')
+        self.url_field.setText('')
+
+        # Add all available plugins to the available plugin list
+        available_plugins = PluginManager.get_all_available()
+        for plugin in available_plugins:
+            self.available_plugin_list.addTopLevelItem(QtWidgets.QTreeWidgetItem([
+                plugin.name(),
+                plugin.version(),
+                plugin.author(),
+                plugin.description()
+            ]))

```

Comment:
This would benefit from the other method, then you could just reset the form in on_start().